subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -0,0 +1,517 @@
+		if (ret >= STATUS_DETAILS_MAX_LEN) {
+			return -ENOMEM;
+		} else if (ret < 0) {
+			return ret;
+		}
+
+		ret =  aws_jobs_update_job_execution(client,
+					      job_id,
+					      state,
+					      status_details,
+					      version_number,
+					      client_token);
+
+		if (ret < 0) {
+			LOG_ERR("aws_jobs_update_job_execution failed: %d",
+				ret);
+		}
+
+		return ret;
+}
+
+
+static int aws_fota_on_publish_evt(struct mqtt_client *const client,
+				   const u8_t *topic,
+				   u32_t topic_len,
+				   u8_t *json_payload,
+				   u32_t payload_len)
+{
+	printk("fota_published_to_sub_topic_evt handler\n");
+	int err;
+
+	/* If not processign job */
+	/* Reciving an publish on notify_next_topic could be a job */
+	if (!strncmp(notify_next_topic, topic,
+		    MIN(NOTIFY_NEXT_TOPIC_MAX_LEN, topic_len))) {
+		err = publish_get_payload(client, payload_len);
+		if (err) {
+			return err;
+		}
+		/*
+		 * Check if the current message recived on notify-next is a
+		 * job.
+		 */
+		printk("Parse document: %s", json_payload);
+		err = aws_fota_parse_notify_next_document(json_payload,
+							  payload_len, job_id,
+							  hostname, file_path);
+		if (err < 0) {
+			printk("Error when parsing the json %d\n", err);
+			return err;
+		}
+
+		/* Unsubscribe from notify_next_topic to not recive more jobs
+		 * while processing the current job.
+		 */
+		err = aws_jobs_unsubscribe_notify_next(client);
+		if (err) {
+			printk("Error when unsubscribing notify_next_topic:"
+				"%d\n", err);
+			return err;
+		}
+
+		/* Subscribe to update topic to recive feedback on wether an
+		 * update is accepted or not.
+		 */
+		err = aws_jobs_subscribe_job_id_update(client, job_id);
+		if (err) {
+			printk("Error when subscribing job_id_update:"
+			       "%d\n", err);
+			return err;
+		}
+
+		/* Construct job_id topics to be used for filtering publish
+		 * messages.
+		 */
+		err = construct_job_id_update_topic(client->client_id.utf8,
+			job_id, "/accepted", job_id_update_accepted_topic);
+		if (err) {
+			printk("Error when constructing_job_id_update_accepted:"
+				"%d\n", err);
+			return err;
+		}
+
+		err = construct_job_id_update_topic(client->client_id.utf8,
+			job_id, "/rejected", job_id_update_rejected_topic);
+		if (err) {
+			printk("Error when constructing_job_id_update_rejected:"
+			       "%d\n", err);
+			return err;
+		}
+
+		/* Set fota_state to DOWNLOAD_FIRMWARE, when we are subscribed
+		 * to job_id topics we will try to publish and if accepted we
+		 * can start the download
+		 */
+		fota_state = DOWNLOAD_FIRMWARE;
+
+		/* Handled by the library*/
+		return 1;
+
+	} else if (!strncmp(job_id_update_accepted_topic, topic,
+			    MIN(JOB_ID_UPDATE_TOPIC_MAX_LEN, topic_len))) {
+		err = publish_get_payload(client, payload_len);
+		if (err) {
+			return err;
+		}
+		doc_version_number++;
+		char rsp_status[STATUS_MAX_LEN + 1] = {0};
+		err = aws_fota_parse_update_job_exec_state_rsp(json_payload,
```

Comment:
`rsp_status` seems to never be read, so what is the purpose of this call?

Reply:
stale, was intended to read out the rsp_status from the received update_job_document but state handling is moved into the library itself as it is the driver of state change.